### PR TITLE
BeginTemporalRo

### PR DESCRIPTION
**Pull request by AskAlexSharov**

### CODE REVIEW

File: cmd/state/exec3/state.go [hunk 1]
```diff
@@ -116,15 +116,15 @@ func (rw *Worker) ResetState(rs *state.StateV3, accumulator *shards.Accumulator)
 	rw.stateWriter = state.NewStateWriterV3(rs, accumulator)
 }
 
-func (rw *Worker) Tx() kv.Tx        { return rw.chainTx }
-func (rw *Worker) DiscardReadList() { rw.stateReader.DiscardReadList() }
+func (rw *Worker) Tx() kv.TemporalTx { return rw.chainTx }
+func (rw *Worker) DiscardReadList()  { rw.stateReader.DiscardReadList() }
 func (rw *Worker) ResetTx(chainTx kv.Tx) {
```
Comment (sudeepdino008) [inline cmd/state/exec3/state.go:121]: makes sense to accept TemporalTx here too

Comment (AskAlexSharov) [reply cmd/state/exec3/state.go:121]: it's rabbit hole - better do by another PR

File: turbo/jsonrpc/otterscan_contract_creator.go [hunk 2]
```diff
@@ -68,7 +68,7 @@ func (api *OtterscanAPIImpl) GetContractCreator(ctx context.Context, addr common
 	}
 
 	var acc accounts.Account
-	ttx := tx.(kv.TemporalTx)
+	ttx := tx
```
[the remark this answers is not in the document]
Comment (AskAlexSharov) [reply turbo/jsonrpc/otterscan_contract_creator.go]: done

**Review by sudeepdino008 — APPROVED** [review]
LGTM apart from failing tests.

**Pull request merged**